doc/source/timeseries.rst

```diff
@@ -889,9 +889,26 @@ The key features of a ``DateOffset`` object are:
 * It can be multiplied by an integer (positive or negative) so that the
   increment will be applied multiple times.
 * It has :meth:`~pandas.DateOffset.rollforward` and
-  :meth:`~pandas.DateOffset.rollback` methods for moving a date forward or 
+  :meth:`~pandas.DateOffset.rollback` methods for moving a date forward or
   backward to the next or previous "offset date".
 
+Beware that ``DatetimeIndexes`` are not assumed to always be monotonic, so
+for example:
+
+.. ipython:: python
+
+df = pd.DataFrame([1,2,3], index=pd.DatetimeIndex(start='2018-02-27',
+                                                  periods=3, freq='D'),
+                                                  columns=['A'])
+df.index
+
+may give you non-monotonic result when performing offsets:
+
+.. ipython:: python
+
+df.index = df.index + pd.tseries.offsets.DateOffset(months=4, days=2)
```

Comment:
what are you trying to show here? this is still monotonic

```
In [3]: (df.index + pd.tseries.offsets.DateOffset(months=4, days=2)).is_monotonic
Out[3]: True

In [4]: df.index.is_monotonic
Out[4]: True
```

Reply:
I think it's a problem of bad choice of wording, what I meant is the output
`DatetimeIndex(['2018-06-29', '2018-06-30', '2018-07-03'], dtype='datetime64[ns]', freq=None)` will skip 2018-07-01 and 2018-07-02 because the days were skip in the original `df.index` 

I will think of a better way to put in words

Reply:
If I'm understanding it correctly, you think there are cases when it's not intuitive that `['2018-01-01', '2018-01-03'] + 1 month` is `['2018-02-01', '2018-02-03']`, and that you could prefer `['2018-02-01', '2018-02-02']` instead?

If that's the case, can you develop a bit on a real-world example? I think that would let us understand better.

I think it may make more sense to move your example to the cookbook, and add a shorter warning here with more information on when that misunderstanding could happen.

Reply:
The problem here is the days that was skipped in February (since it usually end on 28th) does create a couple of days gap in the example above. so I don't understand why you use the beginning of January as an example as it will of cause be intuitive and no confusion.

I am sure there will be real-world example of people falling into the trap as you can refer to the issue #22929

We can move it to the cook book and instead show a link here, no objection on that, if you preferred.

Reply:
Oh, I understand what you meant now. Monotonic means that it never decreases, not that the values are consecutive (I guess consecutive is the best word for this).

I would emphasize that the problem is that adding an offset to an Index with consecutive values can produce an Index with gaps. The example is very helpful, but not immediately obvious, so I'd add the explanation of the problem first.

Also, can you make sure the code in the examples follows PEP-8 please?

Reply:
Oh, I understand what you meant now. Monotonic means that it never decreases, not that the values are consecutive (I guess consecutive is the best word for this).

I would emphasize that the problem is that adding an offset to an Index with consecutive values can produce an Index with gaps. The example is very helpful, but not immediately obvious, so I'd add the explanation of the problem first.

Also, can you make sure the code in the examples follows PEP-8 please?